main/src/ca/uwaterloo/flix/language/ast/TraitUseKind.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2025 Chenhao Gao
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.ast
+
+sealed trait TraitUseKind
+
+object TraitUseKind {
+  case object Implementation extends TraitUseKind
+  case object Derivation extends TraitUseKind
+  case object Constraint extends TraitUseKind
```

Comment:
Sort the above alphabetically and add some comments or examples. Otherwise it will be hard to remember.